src/main/java/net/neoforged/neoforge/flag/DummyFlagManager.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.flag;
+
+import java.util.Set;
+
+final class DummyFlagManager implements FlagManager {
+    private final Set<Flag> enabledFlags;
+
+    public DummyFlagManager(Set<Flag> enabledFlags) {
+        this.enabledFlags = Set.copyOf(enabledFlags);
```

Comment:
The set copy should be a `ReferenceOpenHashSet` to take advantage of the interning of `Flag`s